app/policies/application_policy.rb

```diff
@@ -57,7 +57,32 @@ def initialize(user, scope)
     end
 
     def resolve
+      return scope.all unless user.rbac_enabled?
+
+      graphql_query_by_id? ? graphql_id_query : resolve_scope
+    end
+
+    def resolve_scope
+      raise Exceptions::NotAuthorizedError, "Read access not authorized for #{scope}" unless permission_check('read', scope)
+
       scope.all
     end
+
+    def graphql_id_query
+      id = user.graphql_params.id
+
+      item = scope.find(id)
+      raise Exceptions::NotAuthorizedError, "Read access not authorized for request #{item.id}" unless resource_check('read', item)
+
+      scope.where(:id => id)
```

Comment:
no sure why we need to add `where(:id => id)`? Wont' graphql implementation add it just like graphql_params.filters are handled?

Reply:
`where(:id => id)` will help to populate the return results with their associated resources, ex. every child requests and their actions

Reply:
GraphQL should still work if we don't have rbac enabled. So it lead me to conclude `where(:id=>id)` is not needed.

Reply:
yes, GraphQL should work, but here we only send one record to it, rather than a collection, feel we should have better performance